sumpy/expansion/multipole.py

```diff
@@ -86,10 +86,13 @@ def coefficients_from_source(self, kernel, avec, bvec, rscale, sac=None):
                 rscale, (1,), sac=sac)
 
     def evaluate(self, kernel, coeffs, bvec, rscale, sac=None):
+        from sumpy.tools import DifferentiatedExprDerivativeTaker
         if not self.use_rscale:
             rscale = 1
 
         base_taker = kernel.get_derivative_taker(bvec, rscale, sac)
+        base_taker = DifferentiatedExprDerivativeTaker(base_taker,
```

Comment:
```suggestion
        # Make it so that postprocess_at_target can assume that the
        # "expression" passed to it is either just a sympy expression
        # or a DifferentiatedExprDerivativeTaker.
        base_taker = DifferentiatedExprDerivativeTaker(base_taker,
```

Reply:
(Thanks for also taking a stab at this explanation. Maybe make one that takes the best aspects from both of ours?)